policy-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/policy/common/ContractTestData.java

```diff
@@ -0,0 +1,174 @@
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public static class PassThroughRule<T extends Serializable> implements Rule<T> {
+        @Override
+        public T apply(final T record, final User user, final Context context) {
+            return record;
+        }
+    }
+
+    public static final UserId USER_ID = new UserId().id("test-user-id");
+    public static final UserId NO_RESOURCE_RULES_USER_ID = new UserId().id("noResourceRulesUser");
+    public static final String RESOURCE_ID = "/test/resourceId";
+    public static final String NO_RESOURCE_RULES_RESOURCE_ID = "/test/noRulesResource";
+    public static final String PURPOSE = "test-purpose";
+    public static final String NO_RESOURCE_RULES_PURPOSE = "test-purpose";
+    public static final Context CONTEXT = new Context().purpose(PURPOSE);
+    public static final Context NO_RESOURCE_RULES_CONTEXT = new Context().purpose(NO_RESOURCE_RULES_PURPOSE);
+    public static final User USER = new User().userId(USER_ID).roles("role").auths("auth");
+    public static final User NO_RESOURCE_RULES_USER = new User().userId(NO_RESOURCE_RULES_USER_ID).roles("role").auths("auth");
+    public static final ConnectionDetail CONNECTION_DETAIL = new SimpleConnectionDetail().serviceName("test-data-service");
+    public static final LeafResource RESOURCE = (LeafResource) ResourceBuilder.create(RESOURCE_ID);
+    public static final LeafResource NO_RESOURCE_RULES_RESOURCE = (LeafResource) ResourceBuilder.create(NO_RESOURCE_RULES_RESOURCE_ID);
+    public static final PolicyRequest POLICY_REQUEST;
+    public static final PolicyRequest NO_RESOURCE_RULES_POLICY_REQUEST;
+    public static final JsonNode REQUEST_NODE;
+    public static final JsonNode NO_RESOURCE_RULES_REQUEST_NODE;
+    public static final PolicyRequest REQUEST_OBJ;
+    public static final PolicyRequest NO_RESOURCE_RULES_REQUEST_OBJ;
+    public static final String REQUEST_JSON;
+    public static final String NO_RESOURCE_RULES_REQUEST_JSON;
+
+    static {
+        RESOURCE.connectionDetail(CONNECTION_DETAIL).serialisedFormat("txt").setType("test");
+        POLICY_REQUEST = PolicyRequest.Builder.create()
+                .withUserId(USER_ID.getId())
+                .withResourceId(RESOURCE.getId())
+                .withContext(CONTEXT)
+                .withUser(USER)
+                .withResource(RESOURCE);
+
+        NO_RESOURCE_RULES_RESOURCE.connectionDetail(CONNECTION_DETAIL).serialisedFormat("txt").setType("test");
+        NO_RESOURCE_RULES_POLICY_REQUEST = PolicyRequest.Builder.create()
+                .withUserId(NO_RESOURCE_RULES_USER_ID.getId())
+                .withResourceId(NO_RESOURCE_RULES_RESOURCE.getId())
+                .withContext(NO_RESOURCE_RULES_CONTEXT)
+                .withUser(NO_RESOURCE_RULES_USER)
+                .withResource(NO_RESOURCE_RULES_RESOURCE);
+    }
+
+    static {
+        try {
+            REQUEST_JSON = MAPPER.writeValueAsString(POLICY_REQUEST);
```

Comment:
This should be a hardcoded string for the contract test data, not generated from the above objects

Reply:
updated